cmd/sentry/sentry/sentry_multi_client.go

```diff
@@ -93,20 +98,36 @@ func (cs *MultiClient) RecvMessageLoop(
 		eth.ToProto[eth.ETH66][eth.NewBlockHashesMsg],
 		eth.ToProto[eth.ETH66][eth.NewBlockMsg],
 	}
-	streamFactory := func(streamCtx context.Context, sentry direct.SentryClient) (proto_sentry.Sentry_MessagesClient, error) {
+	streamFactory := func(streamCtx context.Context, sentry direct.SentryClient) (sentryMessageStream, error) {
 		return sentry.Messages(streamCtx, &proto_sentry.MessagesRequest{Ids: ids}, grpc.WaitForReady(true))
 	}
 
-	sentryReconnectAndPumpStreamLoop(ctx, sentry, cs.makeStatusData, "RecvMessage", streamFactory, cs.HandleInboundMessage, wg)
+	sentryReconnectAndPumpStreamLoop(ctx, sentry, cs.makeStatusData, "RecvMessage", streamFactory, makeInboundMessage, cs.HandleAnyInboundMessage, wg)
+}
+
+func (cs *MultiClient) PeerEventsLoop(
+	ctx context.Context,
+	sentry direct.SentryClient,
+	wg *sync.WaitGroup,
+) {
+	streamFactory := func(streamCtx context.Context, sentry direct.SentryClient) (sentryMessageStream, error) {
+		return sentry.PeerEvents(streamCtx, &proto_sentry.PeerEventsRequest{}, grpc.WaitForReady(true))
+	}
+	messageFactory := func() interface{} {
```

Comment:
Candidate for generic?

Reply:
Yes, I guess this can be refactored using generics. Are we already using them elsewhere in the project?

Reply:
no, but we switched to go1.18. only `erigon-lib/common/cmp/cmp.go`

Reply:
improved with generics